source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -293,6 +303,44 @@ void SslSocket::onConnected() { ASSERT(info_->state() == Ssl::SocketState::PreHa
 
 Ssl::ConnectionInfoConstSharedPtr SslSocket::ssl() const { return info_; }
 
+void SslSocket::keylogCallback(const SSL* ssl, const char* line) {
+  ASSERT(ssl != nullptr);
+  BIO* bio_keylog = static_cast<BIO*>(SSL_get_ex_data(ssl, ssl_ex_data_index_));
```

Comment:
I don't think there's a good reason to use the BIO interface for this file IO. Probably using Envoy's access log machinery would be better; it'll get the blocking IO out of the worker thread.

Reply:
keylogCallback will be called in the openssl context. I think it will bring risks to call envoy access log machinery in openssl context.